content/ytt/docs/latest/file-marks.md

```diff
@@ -65,28 +65,47 @@ Change type of file, affecting how `ytt` processes it.
 --file-mark '<path>:type=<file-type>'
 ```
 
-By default `file-type` is determined based on file extension. 
+By default `file-type` is determined based on file extension and — as of v0.32.0 — content.
 
 `file-type` can be: 
-- `yaml-template` (ext: `.yml`) — parsed as a YAML document containing `ytt` templating.
-- `yaml-plain` — parsed as a simple YAML document (not processed for templating).
-- `text-template` (ext: `.txt`) — parsed as a text document containing text templating.
-- `text-plain` — plain text (included in output, as is)
-- `starlark` (ext: `.star`) — a Starlark source file (executed)
-- `data` — a text file that can be loaded by [`data.read()`](lang-ref-ytt.md#data)
- 
+- `yaml-template` (default for: `.yml` or `.yaml`) — parsed as a YAML document and evaluated for templating.
+- `yaml-plain` — parsed as a simple YAML document (_not_ evaluated for templating).
+- `text-template` (default for: `.txt`) — parsed as a text document containing text templating.
+- `text-plain` — included in output, as is.
+- `starlark` (default for: `.star`) — a Starlark source file (executed)
+- `data` (default for all other files) — a text file that can be loaded by [`data.read()`](lang-ref-ytt.md#data)
+
 Example:
 
 ```
 --file-mark 'config.yml:type=data'
 ```
 
-indicates that `config.yml` is _not_ a `yaml-template`, but is `data`. This file will _not_
-be included in the output, but can be loaded using [`data.read()`](lang-ref-ytt.md#data).
+indicates that `config.yml` is _not_ a `yaml-template`, but is `data`. This file will _not_ be parsed, evaluated, or included in the output, but _can_ be loaded using [`data.read()`](lang-ref-ytt.md#data).
+
+#### type detection for YAML files
+_(as of v0.32.0)_
+
+First, `ytt` determines the type of each YAML file:
+1. if it has a `.yml` or `.yaml` extension, it's assumed to be a `yaml-template`
+2. if it is `yaml-template` but does not contain any ytt templating, it is treated as if it were `yaml-plain`
+3. if it is marked with `type=yaml-plain`, it is treated as `yaml-plain`
+
+and then processes each:
```

Comment:
I got lost right here. The above starts with how ytt determines the `type` of a given yaml file, then we get into what happens to them without a transition to lead me.

> First, ytt does XYZ.
yaml-template files are...
yaml-plain files are...


Maybe something as simple as:

First, `ytt` determines the type of each YAML file:
1. X
2. Y
3. Z

Then, once types are determined, ytt can process them as follows:
- yaml-template file are
1. parsed
2. evaluated, and
3. linted

- yaml-plain....


---------------

Is there any other processing that happens for e.g. data files or text-template? Or are yaml files the only ones that get processed? It seems strange that we call these ones out without mentioning the others.